src/api/c/stats.h

```diff
@@ -40,39 +40,52 @@ struct baseOutType {
                                 float>::type type;
 };
 
-template<typename T>
-inline T mean(const Array<T>& in)
+template<typename Ti, typename To>
+inline To mean(const Array<Ti>& in)
```

Comment:
Not these, the other mean which has specilizations. Just did that. Also fixed the covariance unit test that was failing on OSX.